lib/builder.js

```diff
@@ -32,19 +34,27 @@ function summarize(node) {
 RSVP.EventTarget.mixin(Builder.prototype)
 
 Builder.prototype.build = function (willReadStringTree) {
+  if (this.canceled) {
+    return RSVP.Promise.reject(new Error('cannot build this builder, as it has been previously canceled'));
+  }
   var builder = this
 
   var newTreesRead = []
   var nodeCache = []
   var hasMergedInstantiationStack = false;
 
-  return RSVP.Promise.resolve()
+  return (this._currentStep = RSVP.Promise.resolve()
```

Comment:
I'd (ever so slightly) prefer to see the assignment happen before the chaining...